scripts-available/CDB_FederatedServerTables.sql

```diff
@@ -0,0 +1,345 @@
+--------------------------------------------------------------------------------
+-- Private functions
+--------------------------------------------------------------------------------
+
+--
+-- Checks if a column is of integer type
+--
+CREATE OR REPLACE FUNCTION @extschema@.__CDB_FS_Column_Is_Integer(input_table REGCLASS, colname NAME)
+RETURNS boolean
+AS $$
+BEGIN
+    PERFORM atttypid FROM pg_catalog.pg_attribute
+       WHERE attrelid = input_table
+         AND attname = colname
+         AND atttypid IN (SELECT oid FROM pg_type
+           WHERE typname IN
+             ('smallint', 'integer', 'bigint', 'int2', 'int4', 'int8'));
+    RETURN FOUND;
+END
+$$
+LANGUAGE PLPGSQL STABLE PARALLEL UNSAFE;
+
+--
+-- Checks if a column is of geometry type
+--
+CREATE OR REPLACE FUNCTION @extschema@.__CDB_FS_Column_Is_Geometry(input_table REGCLASS, colname NAME)
+RETURNS boolean
+AS $$
+BEGIN
+    PERFORM atttypid FROM pg_catalog.pg_attribute
+        WHERE attrelid = input_table
+           AND attname = colname
+           AND atttypid = '@postgisschema@.geometry'::regtype;
+    RETURN FOUND;
+END
+$$
+LANGUAGE PLPGSQL STABLE PARALLEL UNSAFE;
+
+--
+-- Returns the name of all the columns from a table
+--
+CREATE OR REPLACE FUNCTION @extschema@.__CDB_FS_GetColumns(input_table REGCLASS)
+RETURNS SETOF NAME
+AS $$
+    SELECT
+        a.attname as "colname"
+    FROM pg_catalog.pg_attribute a
+    WHERE
+        a.attnum > 0
+            AND NOT a.attisdropped
+            AND a.attrelid = (
+                SELECT c.oid
+                FROM pg_catalog.pg_class c
+                LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.relnamespace
+                WHERE c.oid = input_table::oid
+            )
+    ORDER BY a.attnum;
+$$ LANGUAGE SQL STABLE PARALLEL UNSAFE;
+
+--
+-- Returns the id column from a view definition
+-- Note: The id is always of one of this forms:
+-- SELECT t.cartodb_id,
+-- SELECT t.my_id as cartodb_id,
+--
+CREATE OR REPLACE FUNCTION @extschema@.__CDB_FS_Get_View_id_column(view_def TEXT)
+RETURNS TEXT
+AS $$
+    WITH column_definitions AS
+    (
+        SELECT regexp_split_to_array(trim(leading from regexp_split_to_table(view_def, '\n')), ' ') AS col_def
+    )
+    SELECT trim(trailing ',' FROM split_part(col_def[2], '.', 2))
+    FROM column_definitions
+    WHERE trim(trailing ',' FROM col_def[array_length(col_def, 1)]) IN ('t.cartodb_id', 'cartodb_id')
+    LIMIT 1;
+$$ LANGUAGE SQL IMMUTABLE PARALLEL SAFE;
```

Comment:
this is not guaranteed to be immutable, is it?

Reply:
It is immutable because it receives a `TEXT` so it'll always do the same.